src/OpenLoco/src/GameCommands/Company/RenameCompanyOwner.cpp

```diff
@@ -108,8 +110,79 @@ namespace OpenLoco::GameCommands
     }
 
     // 0x00434BA1
-    void sub_434BA1()
+    void tryLoadInstalledFaceForName(const CompanyId id, const char* companyName)
     {
-        call(0x00434BA1);
+        const auto competitorInstalledObjects = CompanyManager::findAllOtherInUseCompetitors(id);
+
+        std::optional<ObjectHeader> foundInstalledObject = std::nullopt;
+        for (const auto& object : ObjectManager::getAvailableObjects(ObjectType::competitor))
+        {
+            auto res = std::find(competitorInstalledObjects.begin(), competitorInstalledObjects.end(), object.first);
+            if (res != competitorInstalledObjects.end())
+            {
+                continue;
+            }
+
+            // Copy the string as it needs some processing
+            std::string objectName = object.second._name;
+            // Not sure what 0x88 is but it is at the start of all the object names
+            objectName.erase(std::remove(std::begin(objectName), std::end(objectName), '\x88'));
+
+            auto strcmpSpecial = [](const char* lhs, const char* rhs) {
+                while (*lhs && *rhs)
+                {
+                    const auto chr = *lhs;
+                    if (chr == 0xFF)
+                    {
+                        if (chr != *rhs)
+                        {
+                            return false;
+                        }
+                        lhs++;
+                        rhs++;
+                        const uint16_t unkLhs = *reinterpret_cast<const uint16_t*>(lhs);
+                        const uint16_t unkRhs = *reinterpret_cast<const uint16_t*>(rhs);
+                        if (unkLhs != unkRhs)
+                        {
+                            return false;
+                        }
+                        lhs += 2;
+                        rhs += 2;
+                        continue;
+                    }
+                    auto toLower = [](const char chr) -> char {
```

Comment:
Yeah i know i could have used std::tolower...

Reply:
Any particular reason not to? (or std::toupper, for that matter)

Reply:
because of it being a locale function and all the headaches involved with locale.